apis/v1beta1/validation/gateway.go

```diff
@@ -118,3 +121,39 @@ func ValidateTLSCertificateRefs(listeners []gatewayv1b1.Listener, path *field.Pa
 	}
 	return errs
 }
+
+// ValidateListenerNames validates the names of the listeners
+// must be unique within the Gateway
+func ValidateListenerNames(listeners []gatewayv1b1.Listener, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	nameMap := make(map[gatewayv1b1.SectionName]struct{}, len(listeners))
+	for i, c := range listeners {
+		if _, found := nameMap[c.Name]; found {
+			errs = append(errs, field.Duplicate(path.Index(i).Child("name"), fmt.Sprintln("must be unique within the Gateway")))
```

Comment:
`fmt.Sprintln(<string>)` is a weird construct.

Reply:
Good catch, fixed by #2022.